tensorflow_encrypted/protocol/securenn.py

```diff
@@ -87,8 +94,105 @@ def greater_equal(self, x: PondTensor, y: PondTensor) -> PondTensor:
     def select_share(self, x: PondTensor, y: PondTensor, bit: PondTensor) -> PondTensor:
         return x + bit * (y - x)
 
-    def private_compare(self, x, r, beta):
-        raise NotImplementedError
+    def _private_compare_beta0(self, input: PondPrivateTensor, rho: PondPublicTensor):
+        with tf.device(self.server_0.device_name):
+            wa = 2 * rho * input
+            w = input - wa
+
+            w0_sum = tf.zeros(shape=w.shape)
+            # for i in range(bits-1, -1, -1):
+            # TODO -- subscripting
+            # sum = tf.reduce_sum(w[:, i + 1:], axis=1)
+            # w0_sum[:, i] = sum % p
+
+            c0 = (input * -1) + w0_sum
+
+        with tf.device(self.server_1.device_name):
+            wa = 2 * rho * input
+            wb = rho - wa
+            w = input + wb
+
+            w1_sum = tf.zeros(shape=w.shape)
```

Comment:
maybe you can use [`tf.cumsum`](https://www.tensorflow.org/api_docs/python/tf/cumsum) for this